x-pack/plugins/canvas/public/expression_types/function_form.tsx

```diff
@@ -32,30 +83,37 @@ export class FunctionForm extends BaseForm {
     if (!arg || skipRender) {
       return null;
     }
-
-    const renderArgWithProps = (argValue, valueIndex) =>
+    const renderArgWithProps = (
+      argValue: string | Ast | null,
+      valueIndex: number
+    ): ReactElement<any, any> | null =>
       arg.render({
         key: `${argType}-${expressionIndex}-${arg.name}-${valueIndex}`,
         ...passedProps,
         label,
         valueIndex,
-        argValue,
         onValueChange: onValueChange(arg.name, valueIndex),
         onValueRemove: onValueRemove(arg.name, valueIndex),
+        argValue: argValue ?? null,
       });
 
     // render the argument's template, wrapped in a remove control
     // if the argument is required but not included, render the control anyway
     if (!argValues && arg.required) {
-      return renderArgWithProps({ type: undefined, value: '' }, 0);
+      return renderArgWithProps(null, 0);
     }
 
     // render all included argument controls
-    return argValues && argValues.map(renderArgWithProps);
+    return (
+      argValues &&
+      argValues.map((value, index) => {
+        return renderArgWithProps(value, index);
```

Comment:
can this just be `renderArgWithProps` instead of the function that calls it?

Reply:
Definitely)

Reply:
Updated.